File: The Smallest Difference.cpp
class Solution {
public:
    /**
     * @param A, B: Two integer arrays.
     * @return: Their smallest difference.
     */
    int smallestDifference(vector<int> &A, vector<int> &B) {
        // write your code here
        sort(A.begin(), A.end());
        sort(B.begin(), B.end());
        int i = 0, j = 0;
        int ret = INT_MAX;
        while (i < A.size() && j < B.size())
        {
            ret = min(ret, abs(A[i] - B[j]));
            if (A[i] > B[j])
            {
                ++j;
            }
            else
            {
                ++i;
            }
        }
        return ret;
    }
};

File: The Smallest Difference.py
class Solution:
    # @param A, B: Two lists of integer
    # @return: An integer
    def smallestDifference(self, A, B):
        # write your code here
        
        A.sort()
        B.sort()
        one, two = 0, 0
        
        lenA, lenB = len(A), len(B)
        ans = float('inf')
        while one < lenA and two < lenB:
            ans = min(ans, abs(A[one] - B[two]))
            if A[one] > B[two]:
                two += 1
            else:
                one += 1
        return ans